src/kvstore/raftex/RaftPart.cpp

```diff
@@ -1548,56 +1548,92 @@ void RaftPart::processAppendLogRequest(
     }
 
     // req.get_last_log_id_sent() >= committedLogId_
-    if (lastLogTerm_ > 0 && req.get_last_log_term_sent() != lastLogTerm_) {
-        LOG(INFO) << idStr_ << "The local last log term is " << lastLogTerm_
-                << ", which is different from the leader's prevLogTerm "
-                << req.get_last_log_term_sent()
-                << ", the prevLogId is " << req.get_last_log_id_sent()
-                << ". So need to rollback to last committedLogId_ " << committedLogId_;
-        if (wal_->rollbackToLog(committedLogId_)) {
+    if (req.get_last_log_id_sent() == lastLogId_ && req.get_last_log_term_sent() == lastLogTerm_) {
+        // nothing to do
+        // just append log later
+    } else if (req.get_last_log_id_sent() > lastLogId_) {
+        // There is a gap
+        LOG(INFO) << idStr_ << "Local is missing logs from id " << lastLogId_
+                  << ". Need to catch up";
+        resp.set_error_code(cpp2::ErrorCode::E_LOG_GAP);
+        return;
+    } else  {
+        // check the term
+        int reqLastLogTerm  = wal_->getLogTerm(req.get_last_log_id_sent());
+        if (req.get_last_log_term_sent() != reqLastLogTerm) {
+            LOG(INFO) << idStr_ << "The local log term is " << reqLastLogTerm
+                      << ", which is different from the leader's prevLogTerm "
+                      << req.get_last_log_term_sent()
+                      << ", the prevLogId is " << req.get_last_log_id_sent()
+                      << ". So ask leader to send logs from committedLogId " << committedLogId_;
+            TermID committedLogTerm = wal_->getLogTerm(committedLogId_);
+            if (committedLogTerm > 0 ) {
+                resp.set_last_log_id(committedLogId_);
+                resp.set_last_log_term(committedLogTerm);
```

Comment:
Could consider do not `set_last_log_term`, IIRC, it is not used, so we don't nedd to `wal_->getLogTerm(committedLogId_);`

Reply:
I have considered  it before,  but if we don't set it，then the resp is: (commmitedLogId_,last_log_term). 

 when leader receive the response, it will send a new request  with  (commmitedLogId_,last_log_term) not (commmitedLogId_, commmitedLogIdTerm_), see [  self->lastLogTermSent_ = resp.get_last_log_term();](https://github.com/vesoft-inc/nebula/blob/046b522c2be6bcdd4b27828419c5ae82d21fc844/src/kvstore/raftex/Host.cpp#L310)

append log request with  (commmitedLogId_,last_log_term)  could not pass the check in the follower ,  which will causing a dead loop

Reply:
Yeah, you're right.